### PR TITLE
PTX-1312 fix Panic due to type assertion after k8s API timeout

### PR DESCRIPTION
**Pull request by thiguetta**

### CODE REVIEW

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -1091,10 +1089,8 @@ func (k *K8s) Destroy(ctx *scheduler.Context, opts map[string]bool) error {
 			}
 			return nil, false, nil
 		}
-		pods, err = task.DoRetryWithTimeout(t, k8sDestroyTimeout, DefaultRetryInterval)
-		if err != nil {
-			podList = append(podList, pods.(v1.Pod))
-		}
+
+		task.DoRetryWithTimeout(t, k8sDestroyTimeout, DefaultRetryInterval)
```
Comment (harsh-px) [inline drivers/scheduler/k8s/k8s.go]: we cannot ignore the err here.

File: drivers/scheduler/k8s/k8s.go [hunk 2]
```diff
@@ -1104,10 +1100,8 @@ func (k *K8s) Destroy(ctx *scheduler.Context, opts map[string]bool) error {
...
```
Comment (harsh-px) [inline drivers/scheduler/k8s/k8s.go]: we cannot ignore the err here.

Comment (harsh-px) [reply drivers/scheduler/k8s/k8s.go]: same for other places in PR.

File: drivers/scheduler/k8s/k8s.go [hunk 3]
```diff
@@ -1064,9 +1062,17 @@ func (k *K8s) Destroy(ctx *scheduler.Context, opts map[string]bool) error {
...
-			podList = append(podList, pods.(v1.Pod))
+			// in case we're not waiting for resource cleanup
+			if value, ok := opts[scheduler.OptionsWaitForResourceLeakCleanup]; !ok || !value {
+				return err
+			}
+			if pods != nil {
+				podList = append(podList, pods.([]v1.Pod)...)
+			}
+			// we're ignoring this error since we want to verify cleanup down below, so simply logging it
+			logrus.Errorf("Failed to destroy core objects. Cause: %v", err)
```
Comment (harsh-px) [inline drivers/scheduler/k8s/k8s.go]: if this is an error we are ignoring, use `logrus.Warnf`.

Comment (thiguetta) [reply drivers/scheduler/k8s/k8s.go]: fixed

**Comment by thiguetta** [comment]
jobs passing https://jenkins.portworx.co/job/Torpedo/job/tp-dev-test/111/

**Pull request merged**